src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -620,13 +629,29 @@ private static Expression GetValueFromProperty(Expression sourceExpression, stri
             return Expression.Convert(indexExpression, typeof(string));
         }
 
-        private static Expression BindParameterFromService(ParameterInfo parameter)
+        private static Expression BindParameterFromService(ParameterInfo parameter, FactoryContext factoryContext)
         {
             var isOptional = IsOptionalParameter(parameter);
 
-            return isOptional
-                ? Expression.Call(GetServiceMethod.MakeGenericMethod(parameter.ParameterType), RequestServicesExpr)
-                : Expression.Call(GetRequiredServiceMethod.MakeGenericMethod(parameter.ParameterType), RequestServicesExpr);
+            if (isOptional)
+            {
+                return Expression.Call(GetServiceMethod.MakeGenericMethod(parameter.ParameterType), RequestServicesExpr);
+            }
+            if (factoryContext.ServiceProviderIsService is IServiceProviderIsService serviceProviderIsService)
```

Comment:
If a parameter has the `[FromService]` attribute we should continue to trust it no matter what `IServiceProviderIsService` has to say about it.

Also nit:

```suggestion
            else if (factoryContext.ServiceProviderIsService is IServiceProviderIsService serviceProviderIsService)
```

Reply:
> trust it no matter what `IServiceProviderIsService` has to say about it

If we do that then we would never get the new error below

Reply:
I think that's fine. I don't think `GetRequiredService<{ParameterType}>()` throwing at request-time when a non-optional parameter explicitly attributed with `[FromServices]` is common or hard to diagnose.

I get build-time errors are better than request-time errors, but if `[FromServices]` was actually written out, who are we to say that `IServiceProviderIsService` is more accurate? When `IServiceProviderIsService` was put up for API review with the "DI Council" present, I specifically remember one of the container authors saying they could mostly implement  `IServiceProviderIsService` but there might be some false negatives due to services that get added later or to a scope. I assured them that would be okay because you could always force a parameter to be resolved from the conatiner using the `[FromServices]` attribute.

I really want to try to respect the attributes if at all possible since it's the hammer and easiest behavior to explain.

Reply:
That's fair.